### PR TITLE
improving rules for Bessel functions with half integer indices

### PR DESCRIPTION
**Pull request by mmatera**
This PR improves the rules for Bessel functions of half integer index.

### CODE REVIEW

**Comment by rocky** [comment]
LGTM. 

I am glad to see more code being written in Mathics as opposed to Python as the first choice. 

Also my personal taste would be to write more comments or links to references as was done in some of the discussion in the previous PR. If it wasn't clear back then it might not be clear to someone in the future either.

**Comment by rocky** [comment]
And more doctests showing this aspect off would be nice too, as is done on the WMA site. If we have this goodness in there, we might advertise it a little more.

**Comment by mmatera** [comment]
Here I added a couple of extra rules, including the integral form for BesselJ[0,z]. This can be extended in the future.

File: mathics/autoload/rules/Bessel.m [hunk 1]
```diff
@@ -1,14 +1,79 @@
...
+(*HankelH2[x_Integer?NegativeQ, z_]:=-HankelH2[-x, z];*)
+(*Limit cases*)
+HankelH2[nu_,0] := DirectedInfinity[];
+Protect[HankelH2];
 
 
 Unprotect[BesselI]
-BesselI[1/2,z_] := (Sqrt[2/Pi] Sinh[z]) / Sqrt[z]
-BesselI[-1/2,z_] := (Sqrt[2/Pi] Cosh[z]) / Sqrt[z]
+(*Rayleight's formulas for half-integer indices*)
```
Comment (rocky) [inline mathics/autoload/rules/Bessel.m:20]: Rayleight -> Rayleigh

File: mathics/autoload/rules/Bessel.m [hunk 2]
```diff
@@ -1,14 +1,79 @@
...
+(*Limit cases*)
+BesselI[0, 0] := 1;
+BesselI[nu_Integer,0]:=0;
+BesselI[nu_Rational, 0] := If[nu>0, 0, DirectedInfinity[]];
+BesselI[nu_Real, 0] := If[nu>0, 0, DirectedInfinity[]];
+BesselI[nu_, DirectedInfinity[z___]] := 0;
 Protect[BesselI]
 
+Unprotect[BesselK]
+(*Rayleight's formulas for half-integer indices*)
```
Comment (rocky) [inline mathics/autoload/rules/Bessel.m:32]: eight -> eigh

**Pull request merged**